### DISCUSSION TITLE
Recommendations for powerful code editors?

### DISCUSSION THREAD

**Discussion by jayaddison** [community/community#44157] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Hi folks - I saw some encouragement on my GitHub dashboard to install a powerful code editor.

For the past year or two I've used [`vim`](https://www.vim.org/) for probably 95%+ of my software development work - and can recommend the book ["Mastering Vim"](https://www.packtpub.com/product/mastering-vim/9781789341096) that provides guidance for it.

Can anyone else recommend powerful editors?  Thank you!

![image](https://user-images.githubusercontent.com/55152140/212196379-041ad3b8-7339-4423-9f5d-366f2463439c.png)

(note: I arrived here following the "Product Feedback" link under "Contact GitHub" - please let me know if there's a better place to open this discussion)

**Comment by Chrispulver** [comment]
Yes there is

**Comment by jayaddison** [comment]
Thanks @Chrispulver!

Another family of code editors (that have been around since the 1970s) is [`emacs`](https://en.wikipedia.org/wiki/Emacs) - for example, the widely-used [GNU Emacs](https://www.gnu.org/software/emacs).

The Wikipedia page doesn't describe it as powerful, and I haven't used it much myself, so I couldn't really comment on that.

**Comment by jayaddison** [comment]
Another two code editors that may or may not be powerful include [RStudio](https://www.rstudio.com/) and [Spyder](https://www.spyder-ide.org/).  Both of these cater towards scientific audiences, as I understand it. 

Again, I don't use these myself, so I can't really say much about their abilities, benefits and drawbacks other than to mention that they exist.

**Comment by jayaddison** [comment]
Developers who frequently use SQL databases may find the [Beekeeper Studio](https://www.beekeeperstudio.io/) client and editor environment worth considering.

**Comment by Chrispulver** [comment]
Hello

**Comment by jayaddison** [comment]
[GDevelop](https://editor.gdevelop.io/) and [Ct.js](https://ctjs.rocks/) are programming environments that are designed to assist with computer game development.

**Comment by kvten** [comment, marked as answer]
Depends on what language you're using.

Heard great things about PyCharm (paid, for Python) and Visual Studio (paid, for C++)

**Comment by jayaddison** [comment]
I suppose that emails are rarely _code_ -- except for legal discussions, perhaps.  So I don't know whether this is relevant, but [`mutt`](http://www.mutt.org/) is a free (libre) and open source mail client available on BSD, Linux and Windows (via `cygwin`) platforms.

**Comment by jayaddison** [comment]
> I suppose that emails are rarely code -- except for legal discussions, perhaps. So I don't know whether this is relevant, but [mutt](http://www.mutt.org/) is a free (libre) and open source mail client available on BSD, Linux and Windows (via cygwin) platforms.

That seemed a bit off-topic, to be honest; I guess there's a similarity in the theme of commercial editors (that often seem to rebrand) and free/open/libre editors (of which successful ones appear to last for decades).  In truth there's likely (and beneficially) interplay between those - and for many licenses, there's nothing preventing free/open/libre software from being sold (and/or consulting services provided) for profit.

Anway, I'm not saying anything new there, and there are recommendations for powerful code editors here, as well as me talking to myself (also nothing new), so I'll mark an appropriate answer and close this discussion.

**Comment by jayaddison** [comment]
> (that often seem to rebrand)

Not quite fair either, to be honest - often the branding stays for a long time, even if the product itself is rewritten using entirely different technologies and with entirely different user interfaces.  And if we consider those as two (or more) separate products, then perhaps each is powerful, by some definition -- but are they the same?  Do they use the same formats and provide the same (or an easily transitionary) user interface?  Perhaps not.  Profitable in some cases though, yes.
